docs/archive-data.md

```diff
@@ -0,0 +1,314 @@
+2. **Debugging**: Older data may help with debugging.
+
+## Archival components
+
+Archival consists of four main components.
+
+1. **Archival feature configuration**: Archival is controlled by the [`archival` configuration](https://github.com/temporalio/temporal/blob/master/config/development.yaml#L81) in the `config/development.yaml` file.
+2. **Archive Provider**: Location where the data is archived to. Commonly used providers are S3, GCloud, Kafka, and the local file system.
+3. **Archiver**: System component which archives and retrieves data. An Archiver is directly correlated to a single Archive Provider. 
+4. **URI**: Specifies to the Namespace which Archiver Provider and Archiver should be used. The schema of the URI identifies which Archiver and Provider to use. Where the URI points to identifies the archive location.
+
+## Archival setup    
+
+To setup Archival, there are a few things you will need to do:
+1. Setup archive providers. 
+2. Manage Archival related configurations for the Temporal server.
+3. Create a Namespace that enables Archival and is registered with an Archival URI.
+
+### Archival providers
+
+Temporal supports several archive providers (and archivers) out of the box as well as the option to create and use your own.
+
+To archive data with a specific provider, Temporal must have a corresponding archiver component installed. [Currently supported archive providers](https://github.com/temporalio/temporal/tree/master/common/archiver) include the local filesystem (of whatever host Temporal is currently running on), Google Cloud, and S3.
+
+The local filesystem is used mainly for local installations, testing, and when running Temporal via `docker-compose`. In a production environment, you will most likely want to archive data using one of the cloud based storage systems.
+
+Get started with Google Cloud storage here: https://cloud.google.com/storage
+Get started with S3 storage here: https://aws.amazon.com/s3
+
+You will need the URI of the storage location where you want the data to be archived.
+
+If you want to create your own archiver, review the [Custom archiver](#custom-archiver) section below. Contributions are welcome.
+
+### Archival configuration
+
+First familiarize yourself with the configuration in [`config/development.yaml`](https://github.com/temporalio/temporal/blob/master/config/development.yaml#L81).
+
+| Config |  Description |
+|--------|--------------|
+| `archival.history.state`                       | Acceptable values are `"enabled"`,`"disabled"`. This must be set to `"enabled"` to use the Archival feature with any Namespace in the cluster. |
+| `archival.history.enableRead`                  | Acceptable values are `true`, `false`. This must be set to `true` to read from the archived Event History. |
+| `archival.history.provider`                    | Sub `provider` configs are `filestore`, `gstorage`, `s3`, or `your_custom_provider`. The default config specifies `filestore`. |
```

Comment:
Why is `provider` styled? I think we need to be much more conservative with visual formatting

Reply:
To be honest I struggle with this too but it's really for the best